pkg/controller/deployment/sync.go

```diff
@@ -247,14 +239,40 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 	newRevision := strconv.FormatInt(maxOldRevision+1, 10)
 
 	existingNewRS, err := deploymentutil.FindNewReplicaSet(deployment, rsList)
-	if err != nil {
+	switch {
+	case err != nil:
 		return nil, err
-	} else if existingNewRS != nil {
-		// Set existing new replica set's annotation
+
+	case existingNewRS != nil:
+		// Latest replica set exists. We need to sync its annotations (includes copying all but
+		// annotationsToSkip from the parent deployment, and update revision, desiredReplicas,
+		// and maxReplicas) and also update the revision annotation in the deployment with the
+		// latest revision.
 		if deploymentutil.SetNewReplicaSetAnnotations(deployment, existingNewRS, newRevision, true) {
-			return dc.client.Extensions().ReplicaSets(deployment.ObjectMeta.Namespace).Update(existingNewRS)
+			if _, err := dc.client.Extensions().ReplicaSets(deployment.ObjectMeta.Namespace).Update(existingNewRS); err != nil {
+				return nil, err
+			}
+		}
+
+		updateConditions := false
+		if deployment.Spec.ProgressDeadlineSeconds != nil && len(deployment.Status.Conditions) == 0 {
+			// If no other Conditions have been recorded and we need to estimate the progress of this
+			// deployment then it is likely that old users started caring about progress. In that case
+			// we need to take into account the first time we noticed their new replica set.
+			condition := deploymentutil.NewDeploymentCondition(extensions.DeploymentProgressing, api.ConditionTrue, deploymentutil.FoundNewRSReason, fmt.Sprintf("Found new replica set %q", existingNewRS.Name))
+			deploymentutil.SetDeploymentCondition(&deployment.Status, *condition)
+			updateConditions = true
+		}
+
+		if updateConditions || deploymentutil.SetDeploymentRevision(deployment, newRevision) {
```

Comment:
initialize `updateConditions` with the value `deploymentutil.SetDeploymentRevision(deployment, newRevision)` and remove the or here.